### PR TITLE
Rewrote package using LibSpatialIndex.

### PR DESCRIPTION
**Pull request by evetion**

### CODE REVIEW

**Comment by visr** [comment]
Cool! Did you switch to libspatialindex because you wanted an rtree rather than a kdtree? Did you also consider using https://github.com/alyst/SpatialIndexing.jl?

**Comment by evetion** [comment]
> Cool! Did you switch to libspatialindex because you wanted an rtree rather than a kdtree? Did you also consider using [alyst/SpatialIndexing.jl](https://github.com/alyst/SpatialIndexing.jl)?

Yes, an rtree like structure is the most useful for speeding up aoi queries. The kdtree was more of an experiment to check how this could work, but most applications use extent intersections (mbr).

I did take a look at SpatialIndexing.jl and also considered making several sub-packages (one for each implementation, and this only being an interface package), but found this to be the most quick and pragmatic solution. With the jll you don't notice the C calls at all.

File: src/rtree.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    LibSpatialIndex._checkresult(result, "Couldn't determine bounds of RTree")
+    return Extent(X=(pmins[1], pmaxs[1]), Y=(pmins[2], pmaxs[2]))
+end
+
+function Base.isvalid(R::RTreeIndex)
+    Bool(LibSpatialIndex.C.Index_IsValid(R.I.index))
+end
+
+function Base.findfirst(pred::Base.Fix2{typeof(in),Extent{K,V}}, AA::AcceleratedArray{<:Any,<:Any,<:Any,<:RTreeIndex}) where {N,K,V,T}
+    first(findall(pred, AA))  # this will error on empty sets
```
Comment (visr) [inline src/rtree.jl]: Base returns `nothing` if it cannot be found, perhaps we should do the same?

```
julia> findfirst(==(3), [2,4])

julia> findall(==(3), [2,4])
Int64[]
```

**Review by visr — APPROVED** [review]
Looks good to me!

File: src/rtree.jl [hunk 2]
```diff
@@ -47,7 +47,7 @@ end
 
 function Base.findfirst(pred::Base.Fix2{typeof(in),Extent{K,V}}, AA::AcceleratedArray{<:Any,<:Any,<:Any,<:RTreeIndex}) where {N,K,V,T}
     I = findall(pred, AA)
-    length(I) == 0 ? nothing : first(I)
+    isempty(I) ? nothing : first(I)
```
Comment (evetion) [inline src/rtree.jl:50]: Good catch 👍🏻

**Pull request merged**